crates/swc_ecma_ast/src/expr.rs

```diff
@@ -471,6 +492,102 @@ impl Take for AssignExpr {
     }
 }
 
+impl<'de> Deserialize<'de> for AssignExpr {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct AssignExprVisitor;
+
+        impl<'de> Visitor<'de> for AssignExprVisitor {
+            type Value = AssignExpr;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                formatter.write_str("struct AssignExpr")
+            }
+
+            fn visit_map<M>(self, mut map: M) -> Result<Self::Value, M::Error>
+            where
+                M: MapAccess<'de>,
+            {
+                let mut span_field: Option<Span> = None;
+                let mut op_field: Option<AssignOp> = None;
+                let mut left_field: Option<PatOrExpr> = None;
+                let mut right_field: Option<Box<Expr>> = None;
+
+                while let Some(key) = map.next_key::<&str>()? {
+                    match key {
+                        "span" => {
+                            if Option::is_some(&span_field) {
+                                return Err(de::Error::duplicate_field("span"));
+                            }
+                            span_field = Some(map.next_value()?);
+                        }
+                        "operator" => {
+                            if Option::is_some(&op_field) {
+                                return Err(de::Error::duplicate_field("operator"));
+                            }
+                            op_field = Some(map.next_value()?);
+                        }
+                        "left" => {
+                            if Option::is_some(&left_field) {
+                                return Err(de::Error::duplicate_field("left"));
+                            }
+                            left_field = Some(map.next_value()?);
+                        }
+                        "right" => {
+                            if Option::is_some(&right_field) {
+                                return Err(de::Error::duplicate_field("right"));
+                            }
+                            right_field = Some(map.next_value()?);
+                        }
+                        _ => {
+                            let _: de::IgnoredAny = map.next_value()?;
+                        }
+                    }
+                }
+
+                if span_field.is_none() {
+                    return Err(de::Error::missing_field("span"));
+                }
+                if op_field.is_none() {
+                    return Err(de::Error::missing_field("operator"));
+                }
+                if left_field.is_none() {
+                    return Err(de::Error::missing_field("left"));
+                }
+                if right_field.is_none() {
+                    return Err(de::Error::missing_field("right"));
+                }
+
```

Comment:
Can you use `.ok_or_else` instead for these 4 vars?

Reply:
Done.